File: 10972.py

num=int(input())
a=input().split()
for i in range(len(a)):
    a[i]=int(a[i])
tmp=len(a)-2
b=""
for i in a:
    b=b+str(i)
c=int(b)
b_1=""
bi=a[tmp]
a[tmp]=a[tmp+1]
a[tmp+1]=bi
for i in a:
    b_1=b_1+str(i)
c_1=int(b_1)
if c_1>c:
    for i in range(len(b_1)):
        print(b_1[i],end=" ")
else:
    j=3
    while True:
        if len(a)-j<0:
            print(-1)
            break
        tmp=a[len(a)-j]
        minim=10002
        for i in range(len(a)-j,num):
            if a[i]>tmp and a[i]<minim:
                minim=a[i]
                minim_idx=i
        if minim==10002:
            j=j+1
            continue
        else:
            bi=a[len(a)-j]
            a[len(a)-j]=a[minim_idx]
            a[minim_idx]=bi
            res=""
            for i in range(len(a)-j+1,num):
                res=res+str(a[i])
            resu=list(res)
            for i in range(len(resu)):
                resu[i]=int(resu[i])
            resu.sort()
            for i in range(len(a)-j+1,num):
                a[i]=resu[i-len(a)+j-1]
            for i in a:
                print(str(i),end=" ")
            break
            

File: 10972.cpp
#include<iostream>
#include<stdlib.h>
using namespace std;
void sorting(int *a,int n){
	int i,j,tmp;
	for(i=0;i<n-1;i++){
		for(j=i+1;j<n;j++){
			if(a[i]>a[j]){
				tmp=a[i];
				a[i]=a[j];
				a[j]=tmp;
			}
		}
	}
}
int main(void){
	int n;
	int arr[10000];
	cin>>n;
	for(int i=0;i<n;i++){
		cin>>arr[i];
	}
	if(n==1){
		cout<<"-1"<<endl;
		return 0;
	}
	int tmp=n-2;
	int bi=arr[tmp];
	if(arr[tmp]<arr[tmp+1]){
		int bi=arr[tmp];
		arr[tmp]=arr[tmp+1];
		arr[tmp+1]=bi;
		for(int i=0;i<n;i++){
			cout<<arr[i]<<" ";
		}
		cout<<endl;
	}
	else{
		int j=3;
		while(true){
			if(n-j<0){
				cout<<"-1"<<endl;
				break;
			}
			tmp=arr[n-j];
			int minim=10002;
			int minim_idx;
			for(int i=n-j+1;i<n;i++){
				if(arr[i]>tmp&&arr[i]<minim){
					minim=arr[i];
					minim_idx=i;
				}
			}
			if(minim==10002){
				j=j+1;
				continue;
			}
			else{
				bi=arr[n-j];
				arr[n-j]=arr[minim_idx];
				arr[minim_idx]=bi;
				int *res;
				res=(int*)malloc(sizeof(int)*(j-1));
				for(int i=n-j+1;i<n;i++){
					res[i-n+j-1]=arr[i];
				}
				sorting(res,j-1);
				for(int i=n-j+1;i<n;i++){
					arr[i]=res[i-n+j-1];
				}
				for(int i=0;i<n;i++){
					cout<<arr[i]<<" ";
				}
				cout<<endl;
				break;
			}
		}
	}
	return 0;
}
